internal/service/streamprocessor/data_source_schema.go

```diff
@@ -0,0 +1,61 @@
+package streamprocessor
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+)
+
+func DataSourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				Description:         "Unique 24-hexadecimal character string that identifies the stream processor.",
+				MarkdownDescription: "Unique 24-hexadecimal character string that identifies the stream processor.",
+			},
+			"instance_name": schema.StringAttribute{
+				Required:            true,
+				Description:         "Human-readable label that identifies the stream instance.",
+				MarkdownDescription: "Human-readable label that identifies the stream instance.",
+			},
+			"pipeline": schema.StringAttribute{
+				Computed:            true,
+				Description:         "Stream aggregation pipeline you want to apply to your streaming data.",
+				MarkdownDescription: "Stream aggregation pipeline you want to apply to your streaming data.",
+			},
+			"processor_name": schema.StringAttribute{
+				Required:            true,
+				Description:         "Human-readable label that identifies the stream processor.",
+				MarkdownDescription: "Human-readable label that identifies the stream processor.",
+			},
+			"project_id": schema.StringAttribute{
+				Required:            true,
+				Description:         "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
```

Comment:
Should id be capitalized: ID?

Reply:
`id` is used consistently throughout the API so I think we should keep it as is.